src/qutip_qoc/_rl.py

```diff
@@ -189,10 +181,8 @@ def step(self, action):
 
         for i, value in enumerate(alphas):
             self.args[f"alpha{i+1}"] = value
-        self._H = qt.QobjEvo(self._H_lst, self.args)
 
-        self.update_solver()                # _H has changed
-        infidelity = self.infidelity()
+        infidelity = self._infid()
```

Comment:
`params` argument is not used.
Instead of storing them in a class attribute, I would suggest to pass them to the function.